### PR TITLE
more C352 improvements

### PR DESCRIPTION
**Pull request by superctr**
- implement volume ramping behavior and confirmed with recording of real hardware (example: Tekken 3, reduced pops in especially King's stage BGM)
- rewrite mulaw algorithm. This might not be quite perfect yet but is closer to recordings and sounds a bit clearer (example: Time Crisis)
- byte swap sample ROMs for adillor and primglex.

### CODE REVIEW

File: src/mame/drivers/namcos11.cpp [hunk 1]
```diff
@@ -1188,7 +1188,7 @@ ROM_START( primglex )
 	ROM_LOAD( "pg1sprog.6d",  0x0000000, 0x040000, CRC(e7c3396d) SHA1(12bbb8ebcaab1b40462a12917dd9b58bd9ab8663) )
 
 	ROM_REGION( 0x1000000, "c352", 0 ) /* samples */
-	ROM_LOAD( "pg1wave.8k",   0x0000000, 0x400000, CRC(fc9ad9eb) SHA1(ce5bb2288ed8cf1348825c39423cbb99d9324b9c) )
+	ROM_LOAD( "pg1wave.8k",   0x0000000, 0x400000, CRC(67389b6a) SHA1(43c0f89378eb1dcb9a159a19e9ba4f8985a85846) )
```
Comment (rb6502) [inline src/mame/drivers/namcos11.cpp]: Do not modify the ROMs, use ROM_LOAD16_WORD_SWAP().

File: src/mame/drivers/namcos22.cpp [hunk 2]
```diff
@@ -5388,8 +5388,8 @@ ROM_START( adillor )
 	ROM_LOAD( "am1ptru3.14f", 0x580000, 0x080000, CRC(5e4e6333) SHA1(5897251e4694c5c24f1810ff6f9177a0456baf2e) )
 
 	ROM_REGION( 0x1000000, "c352", 0 ) /* sound samples */
-	ROM_LOAD( "am1wavea.2l",  0x000000, 0x400000, CRC(48f8c20c) SHA1(48b4fbcb7e9dbbb70a542ef7cb7ee0e46fad23fc) )
-	ROM_LOAD( "am1waveb.1l",  0x800000, 0x400000, CRC(fd8e7384) SHA1(91e53ab0293f81f8357645fd319249abc128b78e) )
+	ROM_LOAD( "am1wavea.2l",  0x000000, 0x400000, CRC(2dd64531) SHA1(255a44cb5cda2a76a8c557ef242295d7290712aa) )
```
Comment (rb6502) [inline src/mame/drivers/namcos22.cpp]: Same here; use ROM_LOAD16_WORD_SWAP() instead of modifying the ROMs.

**Comment by superctr** [comment]
I modified the ROMs because they should have been byteswapped to begin with, the bus is 8-bit. There is no rom decode magic happening either. the edit is just a regular "dd conv=swab".

**Comment by happppp** [comment]
I also remember this one:
https://github.com/superctr/mame/commit/2dabc32063caf01c6758e96c24b8a5
8f822223c4

Should we revert it and load with word swap?
superctr: can you doublecheck that ts1waveb rom is ok?

**Comment by superctr** [comment]
ts1waveb is OK. It contains all the voice samples and some sound effects, which sound OK.

I do want to stress that I think that there is something wrong with some the dumps. Another strange case is Dunk Mania (dm1wave.8k). It seems like every even or odd byte has been duplicated.

**Comment by p1pkin** [comment]
@happppp  I think so. according to documentation there used 2 different type mask roms in different modes. I'm pretty sure Guru had enough skills to dump this properly.
`` WAVEA.2L   WAVEB.1L    : 16M/32M WAVE MASKROMs. If 32MBit DIP42, they're programmed in BYTE mode (DIP42/SOP44)
``Time Crisis        'TS1 WAVE A', 'TS1 WAVE B' , WAVE A DIP42 32M MASKROM, WAVE B DIP42 16M MASKROM

@superctr it seems you not really understand how things implemented in hardware, its pretty common to have data byte order swapped, or even individual bits, interleaved in various ways, etc.

**Comment by superctr** [comment]
I do know how things are implemented in hardware, and I don't deny that it's possible to swap the byte order. But I also know from reversing the chip's pinout that the data bus is 8-bit. And that it is just strange that some dumps have swapped byte order even though other games (even using the same ROM board) have the correct byte order (that is possible if the mask ROMs are 8/16-bit and the dumps were just done in word mode). I just suggested that the byte order in the dumps should be kept consistent.

**Pull request merged**